### ISSUE TITLE
[web] Service worker is caching CORE resources on install but they are never used

### ISSUE THREAD

**Issue by azbuky** [flutter/flutter#68953]
When deploying a Flutter web project in production, when the service worker installs in the users web browser, it caches the CORE resources with the '?revision=#hash' parameter. These cached files are never used with the revision hash. As a result main.dart.js is requested and cached twice.

## Steps to Reproduce

1. Run `flutter create .`
2. Run `flutter build web`
3. Deploy in a hosting environment. (e.g. Run `firebase deploy` after configuring hosting)

**Expected results:**

CORE files (e.g. main.dart.js) are only loaded once.

**Actual results:** 
Chrome loads the following resources after first page load
![Loaded resources in Chrome on first page load](https://user-images.githubusercontent.com/567277/97082781-eb6edb80-1614-11eb-8a9e-468218a27b0d.png)

Requests on the server side
![Requests on the server side](https://user-images.githubusercontent.com/567277/97082798-017c9c00-1615-11eb-8475-2066de54d81c.png)

Cached resources (by the service worker) after first page load
![Cached resources (by the service worker) after first page load](https://user-images.githubusercontent.com/567277/97082807-148f6c00-1615-11eb-92df-a47f30b42a95.png)

Loaded resources after second page load (note that main.dart.js is loaded by the service worker from the browser's disk cache)
![Loaded resources after second page load ](https://user-images.githubusercontent.com/567277/97082866-6801ba00-1615-11eb-83a6-9e53ceae5af3.png)

Cached resources (by service worker) after second page load
![Cached resources after second page load](https://user-images.githubusercontent.com/567277/97082841-3daffc80-1615-11eb-9cb9-a5aa95a5d474.png)

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.24.0-4.0.pre.51, on Linux, locale en_US.UTF-8)
    • Flutter version 1.24.0-4.0.pre.51 at /home/azbuky/fvm/versions/master
    • Framework revision 720dff6a94 (6 hours ago), 2020-10-24 03:57:03 -0400
    • Engine revision 432974a52b
    • Dart version 2.11.0 (build 2.11.0-251.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/azbuky/Devel/Sdks/Android/Sdk/
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /home/azbuky/Devel/Sdks/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio
    • Android Studio at /home/azbuky/Devel/Sdks/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /home/azbuky/Devel/Sdks/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.15.1

[✓] Connected device (3 available)
    • Linux (desktop)  • linux      • linux-x64      • Linux
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 86.0.4240.111

• No issues found!
```

</details>

**Comment by TahaTesser** [comment]
Hi @azbuky 
I see there's an open issue addressing the case you described. https://github.com/flutter/flutter/issues/55967
Please follow up on that issue, I'm closing the current one as a duplicate.
If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by azbuky** [comment]
Hi @TahaTesser 
That issue that you mention is quite old and there have been quite a few changes to the service worker since then like #64976 and #66069 that were supposed to fix #64968

There is another report from @kmcgill88 [describing](https://github.com/flutter/flutter/issues/64968#issuecomment-706270230) this same issue I encounter.

@jonahwilliams please take a look at this when you have time.

**Comment by TahaTesser** [comment]
Hi @azbuky 
Its an old issue but it is meant to make service worker caching smarter, the points you mentioned could be added there 
If @jonahwilliams thinks otherwise, we can reopen this one
